Feature/task 10 restructure and improv

JAmmermann-DLR opened the pull request:
- Update antscript:
  - runs on Windows and Unix
  - on Windows it will extract FreeCAD into the folder _FreeCAD_
  - on Unix:
    - FreeCAD gets extracted from an _AppImage_ into the folder _squashfs-root_, which can't be moved or renamed because of dependencies
    - the extraction has to be done because a python file has to be patched to make the testcases pass
    - in future releases of FreeCAD this step could be avoided
- Update travis:
  - now uses the antscript to install FreeCAD
  - now only deploys the _VirtualSatelliteCAD_ folder

Commit:
Added new module folder for restructuring project - (Task #10)

- Added Virtual Satellite folder
- Adjusted python build path to new folder
- Moved all code to new folder
- Adjusted Launcher for Module Tests

---
Task #10: Restructure and improve project setup

Commit:
Added FreeCAD folder with .freecad - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Updated .gitignore - (Task #10)

Ignore FreeCAD folder contents

---
Task #10: Restructure and improve project setup

Commit:
Initial ant script for setup - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Added libs for unzipping z7ip from ant - (Task #10)

- Added antlib-compress from apache (Apache License 2.0)
- Added commons-compress from apache (Apache License 2.0)
- Added XZ-java (Public Domain)

---
Task #10: Restructure and improve project setup

Commit:
Update test .launch configurations - (Task #10)

Replace static paths with workspace relative paths.

---
Task #10: Restructure and improve project setup

Commit:
Unzipping FreeCAD without creating a new subfolder- (Task #10)


---
Task #10: Restructure and improve project setup

Commit:
Add FreeCAD patch - (Task #10)

- Add target to copy patched file to FreeCAD dir
- Refactor properties

---
Task #10: Restructure and improve project setup

Commit:
Add A2plus to build file- (Task #10)

download and unzip a2plus

---
Task #10: Restructure and improve project setup

Commit:
Update build file for differnt os and first travis test- (Task #10)

build.xml:
- different handling depending on os
- add dependencies between targets
- add main target to call all targets in the right order

travis.yml:
- try to comment out unneeded parts (e.g. freecad and a2plus
installation)
- try to call the ant build.xml file

---
Task #10: Restructure and improve project setup

Commit:
Update ant call in travis.yml - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Fix travis tests (freecad and flake) - (Task #10)

Try to replace freecad with call to FreeCAD .exe created by ant script
Try to run flake on the VirtualSatelliteCAD directory only

---
Task #10: Restructure and improve project setup

Commit:
Update FreeCAD Linux download link in build.xml - (Task #10)

- provide the right link for Linux to the AppImage
- download it
- make it executable
- use it in travis.yml

---
Task #10: Restructure and improve project setup

Commit:
Add missing if clauses for OS in build.xml - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Extract the FreeCAD AppImage in build.xml - (Task #10)

- run the AppImage with "--appimage-extract" and then move the output in
the wanted directory
- make FreeCAD executable
- adapt call in travis

---
Task #10: Restructure and improve project setup

Commit:
Fix typo in build.xml - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Get travis debug output in build.xml - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Find location of squashfs-root directory - (Task #10)

Current error caused by squashfs-root not being at the expected
location, so debug output was added
---
Task #10: Restructure and improve project setup

Commit:
Correct squashfs-root fs location - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Fix ant target for All Tests in travis - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Fix bug when patching FreeCAD file - (Task #10)

Didn't overwrite existing files, added overwrite attribute.
Also commented out All Test because of long runtime and errors to be
investigated later.

---
Task #10: Restructure and improve project setup

Commit:
Update location dir for TestVirtualSatelliteApp in travis - (Task #10)

Also comment out second build for faster execution time

---
Task #10: Restructure and improve project setup

Commit:
Reactivate all tests in travis - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Add missing packages for all FreeCAD tests - (Task #10)

Try to solve error with libfreeimage3

---
Task #10: Restructure and improve project setup

Commit:
Load shared libraries with ldconfig- (Task #10)

A couple of shared libraries seem to not be available, so try to reload
them implicit with ldconfig

---
Task #10: Restructure and improve project setup

Commit:
Call ldconfig as sudo - (Task #10)

Fix permission denied error

---
Task #10: Restructure and improve project setup

Commit:
Add libjsoncpp-dev package - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Resolve error with needed packages by setting env variable - (Task #10)

Try to resolve "xkbcommon: ERROR: failed to add default include path
[...]/usr/lib" that is probably caused the pyqt gui. So try to set the
corresponding environment variable manual"
---
Task #10: Restructure and improve project setup

Commit:
Update QT_XKB_CONFIG_ROOT environment variable - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Absolute path for QT_XKB_CONFIG_ROOT - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Refactor ant script for handling different OS - (Task #10)

The needed dependencies for running the FreeCAD test on Linux are
resolved by the AppRun, thus the folder structure of the extracted
AppImage has to stay intact. So we have to completely differentiate
between Windows and Linux in the ant script.

---
Task #10: Restructure and improve project setup

Commit:
Update wrong property - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Rename squashfs-root to FreeCAD folder - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Update if clause for unix targets - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Update AppRun destination - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Dont't set QT_XKB_CONFIG_ROOT - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Don't move squashfs-root - (Task #10)

This caused the problems with shared libraries not found.

---
Task #10: Restructure and improve project setup

Commit:
Cleanup build and travis - (Task #10)

Remove unneeded comments.

---
Task #10: Restructure and improve project setup

Commit:
Update deploy folder - (Task #10)

Only deploy "VirtualSatelliteCAD".

---
Task #10: Restructure and improve project setup

PhilMFischer commented on build.xml:
```diff
@@ -0,0 +1,130 @@
+		</classpath>
+	</taskdef>
+
+	<target name="Main" depends="Cleanup">
```
Can we maybe call it differently?
Maybe we should also put a bit explanation into the script, so that developers know what this is all about.

JAmmermann-DLR replied:
Resolved the issues in https://github.com/virtualsatellite/VirtualSatellite4-FreeCAD-mod/pull/16/commits/0edc586b0228cd9847ac23e2f8e1edab5439edcd:

- Added the zips for the ant script and removed documentation file
- Refactored the build.xml
- Updated the Readme.md to mention the ant script and third party dependencies

Commit:
Refactor antlib directory - (Task #10)

- Add zips
- Remove documentation files

---
Task #10: Restructure and improve project setup

Commit:
Refactor ant script - (Task #10)

- Update the default target
- Split Cleanup target into OS dependent targets
- Add comment for the ant script workflow

---
Task #10: Restructure and improve project setup

Commit:
Update default target name in build.xml and travis.yml - (Task #10)

---
Task #10: Restructure and improve project setup

Commit:
Update README.md

- Add ant script to section "Quickstart for developers"
- Add section "Third party dependencies"

Commit:
Adjusted Readme - (Task #10)



---
Task #10: Restructure and improve project setup

PhilMFischer commented:
Closes #10
